jsonschema-core.xml

```diff
@@ -800,49 +800,302 @@
+                    The URI MAY be a URL, but the nature of the retrievable resources is
+                    currently undefined, and reserved for future use.  Vocabulary authors
+                    SHOULD NOT serve a document at that URL.  A server MAY respond with
+                    the relevant protocol's successful "no content" message, such as
+                    an HTTP 204 status.
+                    <cref>
+                        Vocabulary documents may be added shortly, or in the next draft.
+                        For now, identifying the keyword set is deemed sufficient as that,
+                        along with meta-schema validation, is how the current "vocabularies"
+                        work today.
+                    </cref>
+                </t>
+                <t>
+                    The values of the object properties MUST be booleans.
+                    If the value is true, then implementations that do not recognize
+                    the vocabulary MUST refuse to process any schemas that declare
+                    this meta-schema with "$schema".  If the value is false, implementations
+                    that do not recognize the vocabulary MAY choose to proceed with processing
+                    such schemas.
+                </t>
+                <t>
+                    When processing a schema that uses unrecognized vocabularies, keywords
+                    declared by those vocabularies are treated like any other unrecognized
+                    keyword, and ignored.
+                </t>
+                <t>
+                    The "$vocabulary" keyword SHOULD be used in the root schema of any schema
+                    document intended for use as a meta-schema.  It MUST NOT appear in subschemas.
+                </t>
+                <t>
+                    The "$vocabulary" keyword MUST be ignored in schema documents that
+                    are not being processed as a meta-schema.  This allows validating
+                    a meta-schema M against its own meta-schema M' without requiring
+                    the validator to understand the vocabularies declared by M.
+                </t>
+                <t>
+                    Note that the processing restrictions on "$vocabulary" mean that
+                    meta-schemas that reference other meta-schemas using "$ref" or
+                    similar keywords do not automatically inherit the vocabulary
+                    declarations of those other meta-schemas.  All such declarations
+                    must be repeated in the root of each schema document intended
+                    for use as a meta-schema.  This is demonstrated in
+                    <xref target="example-meta-schema">the example meta-schema</xref>.
+                </t>
+                <t>
+                    If "$vocabulary" is absent, an implementation MAY determine
```

Comment:
What would happen if a schema declares `"$schema": "file://draft-07/schema"` and that meta-schema is just the plain draft-07 schema (properly identified by its `$id`)? Wouldn't it be also useful for implementations to let them determine the behavior based on the meta-schema own identifier?

Reply:
I don't mean to force implementations to do it, but rather let them optionally do so, in addition to or instead of doing it based on the schema's `$schema`. 

I was just thinking that with the shift of paradigm in this draft, moving the vocabulary declaration place from the schema to the meta-schema, some implementations might prefer to always perform the vocabulary determination during meta-schema processing, not having anything hardcoded, and therefore it could be more convenient and straightforward, when a previous draft meta-schema is encountered, to fallback to recognize the vocabulary based on a meta-schema property instead of information found on a "different processing layer".

Reply:
A possible wording for my proposal could be:
> If "$vocabulary" is absent, an implementation MAY determine behavior based on the meta-schema if it is recognized from the URI value of the referring schema's "$schema" keyword. Additionally, an implementation MAY determine behavior based on the meta-schema if it is recognized from the URI value of the meta-schema's "$id" keyword. If the meta-schema is not recognized, then implementations MAY assume the use of all vocabularies in this specification and the companion Validation specification.

Reply:
@jgonzalezdr We're going to stick with exactly one unambiguous mechanism for now, which matches the behavior we have always had with `$schema`.  If we get feedback from the new draft that a change is needed, we'll listen to that.

Reply:
It's fine then.